oap-server/server-alarm-plugin/src/main/java/org/apache/skywalking/oap/server/core/alarm/provider/RunningRule.java

```diff
@@ -383,6 +392,14 @@ private boolean isMatch() {
             return isMatch == 1;
         }
 
+        public boolean checkOutOfDate(LocalDateTime currentTime) {
+            if (this.lastUpdateTime != null) {
+                int minutes = Minutes.minutesBetween(lastUpdateTime, currentTime).getMinutes();
+                return minutes > this.period * 10;
+            }
+            return false;
+        }
```

Comment:
If the window of this entity is out of date, shouldn't all `values` are null? I think it is better compared with this condition.

Reply:
Indeed